lib/cadet/assessments/assessments.ex

```diff
@@ -448,59 +449,40 @@ defmodule Cadet.Assessments do
           assessment: a
       })
 
-    case role do
-      @grading_role ->
-        students = Cadet.Accounts.Query.students_of(grader)
+    cond do
+      role in @grading_roles ->
+        {:ok, submissions_by_group(grader, submission_query)}
 
+      role in @see_all_submissions_roles ->
         submissions =
-          submission_query
-          |> join(:inner, [s], st in subquery(students), s.student_id == st.id)
-          |> Repo.all()
+          (group_only && role != @admin_role && submissions_by_group(grader, submission_query)) ||
+            Repo.all(submission_query)
 
         {:ok, submissions}
 
-      @see_all_submissions_role ->
-        submissions = Repo.all(submission_query)
-
-        {:ok, submissions}
-
-      _ ->
+      true ->
         {:error, {:unauthorized, "User is not permitted to grade."}}
     end
   end
 
   @spec get_answers_in_submission(integer() | String.t(), %User{}) ::
           {:ok, [%Answer{}]} | {:error, {:unauthorized, String.t()}}
-  def get_answers_in_submission(id, grader = %User{role: role}) when is_ecto_id(id) do
+  def get_answers_in_submission(id, %User{role: role}) when is_ecto_id(id) do
     answer_query =
       Answer
       |> where(submission_id: ^id)
       |> join(:inner, [a], q in assoc(a, :question))
       |> preload([a, q], question: q)
 
-    case role do
-      @grading_role ->
-        students = Cadet.Accounts.Query.students_of(grader)
-
-        answers =
-          answer_query
-          |> join(:inner, [a], s in Submission, a.submission_id == s.id)
-          |> join(:inner, [a, _, s], t in subquery(students), t.id == s.student_id)
-          |> Repo.all()
-          |> Enum.sort_by(& &1.question.display_order)
-
-        {:ok, answers}
-
-      @see_all_submissions_role ->
-        answers =
-          answer_query
-          |> Repo.all()
-          |> Enum.sort_by(& &1.question.display_order)
-
-        {:ok, answers}
+    if role in @see_all_submissions_roles do
```

Comment:
You'll need to restore the old one and use
```elixir
cond do
  role in @grading_roles ->
    ...
  role in @see_all_submission_roles ->
    ...
  true ->
    ...
```

Reply:
Ahh yup! Missed this out

Done!